src/PowerShellEditorServices/Analysis/AnalysisService.cs

```diff
@@ -467,25 +495,85 @@ private PSObject[] InvokePowerShell(string command, IDictionary<string, object>
+        /// This uses the PowerShell module API to load the first PSScriptAnalyzer module on the module path with
+        /// with the required minimum version and then creates a runspace pool with that module loaded in the initial
+        /// session state.
+        /// </summary>
+        /// <returns>A runspace pool with PSScriptAnalyzer loaded for running script analysis tasks.</returns>
+        private static RunspacePool CreatePssaRunspacePool(out PSModuleInfo pssaModuleInfo)
         {
             using (var ps = System.Management.Automation.PowerShell.Create())
             {
+                // Run `Get-Module -ListAvailable -Name "PSScriptAnalyzer"`
                 ps.AddCommand("Get-Module")
                     .AddParameter("ListAvailable")
                     .AddParameter("Name", PSSA_MODULE_NAME);
 
                 try
                 {
-                    return ps.Invoke()?.Any() ?? false;
+                    // Get the latest version of PSScriptAnalyzer we can find
+                    pssaModuleInfo = ps.Invoke()?
+                        .Select(psObj => psObj.BaseObject)
+                        .OfType<PSModuleInfo>()
+                        .OrderBy(moduleInfo => moduleInfo.Version)
+                        .FirstOrDefault();
                 }
-                catch (Exception)
+                catch (Exception e)
                 {
-                    return false;
+                    throw new Exception("Unable to find PSScriptAnalyzer module", e);
                 }
+
+                if (pssaModuleInfo == null)
+                {
+                    throw new Exception("Unable to find PSScriptAnalyzer module");
+                }
+
+                // Create a base session state with PSScriptAnalyzer loaded
+                InitialSessionState sessionState = InitialSessionState.CreateDefault2();
+                sessionState.ImportPSModule(new [] { pssaModuleInfo.ModuleBase });
+
+                // RunspacePool takes care of queuing commands for us so we do not
+                // need to worry about executing concurrent commands
+                return RunspaceFactory.CreateRunspacePool(sessionState);
             }
         }
 
         #endregion //private methods
+
+        #region IDisposable Support
+
+        private bool _disposedValue = false; // To detect redundant calls
+
+        /// <summary>
+        /// Dispose of this object.
+        /// </summary>
+        /// <param name="disposing">True if the method is called by the Dispose method, false if called by the finalizer.</param>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!_disposedValue)
+            {
+                if (disposing)
+                {
+                    _analysisRunspacePool.Close();
+                    _analysisRunspacePool.Dispose();
+                    _analysisRunspacePool = null;
```

Comment:
I'm pretty sure you only need to do the Close, here. I think the Close does a Dispose. I'm also curious if the null line is needed.

Reply:
Had a quick look at the code and it looks like `Dispose()` calls `Close()`:
https://github.com/PowerShell/PowerShell/blob/c1c5344a8897262433141ecbc13bb06ac2c4bbef/src/System.Management.Automation/engine/hostifaces/RunspacePoolInternal.cs#L823-L836

I'm not sure that `Close()` calls `Dispose()`.

Dunno how many resources or how much memory a runspace pool allocates, but the comments generated with the Dispose pattern snippet indicate large resources [should be set to `null`](https://gist.github.com/traktraktrugui/7330187#file-bdisp-xml-L46-L47).